src/spec-configuration/containerFeaturesConfiguration.ts

```diff
@@ -225,6 +225,54 @@ USER $_DEV_CONTAINERS_IMAGE_USER
 `;
 }
 
+function getFeatureInstallWrapperScriptTemplate() {
+    return `#!/bin/sh
+set -e
+
+on_exit () {
+	[ $? -eq 0 ] && exit
+	echo 'ERROR: Feature "#{name}" (#{id}) failed to install! Look at the documentation at #{documentation} for help troubleshooting this error.'
+}
+
+trap on_exit EXIT
+
+echo ===========================================================================
+echo 'Feature       : #{name}'
+echo 'Description   : #{description}'
+echo 'Id            : #{id}'
+echo 'Version       : #{version}'
+echo 'Documentation : #{documentation}'
+echo 'Options       :'
+echo '#{options}'
+echo ===========================================================================
+
+set -a
+. ./devcontainer-features.env
+set +a
+
+chmod +x ./install.sh
+./install.sh
+`;
+}
+
+export function getFeatureInstallWrapperScript(feature: Feature, featureSet: FeatureSet, options: string[]): string {
+	return getFeatureInstallWrapperScriptTemplate()
```

Comment:
You could use the `${expression}` syntax inside the above template string to simplify the replacement.

Reply:
Good call. We still need to do the quote escaping, but I think it will be cleaner with the string interpolation. I'll play around with it and see how it looks.

Reply:
Updated to use string interpolation. I agree it's cleaner this way, and it actually makes it easier to customize the header a bit more when some optional properties are undefined. 👍